src/entities/User/routes.ts

```
@@ -0,0 +1,125 @@
+  const message = {
+    address,
+    timestamp: new Date().toISOString(),
+  }
+
+  const messageTimeout = setTimeout(() => {
+    delete VALIDATIONS_IN_PROGRESS[address]
+  }, MESSAGE_TIMEOUT_TIME)
+
+  VALIDATIONS_IN_PROGRESS[address] = {
+    ...message,
+    messageTimeout,
+  }
+
+  return formatValidationMessage(address, message.timestamp)
+}
+
+async function checkValidationMessage(req: WithAuth) {
+  const user = req.auth!
+  try {
+    const messageProperties = VALIDATIONS_IN_PROGRESS[user]
+    if (!messageProperties) {
+      throw new Error('Validation timed out')
+    }
+
+    const { address, timestamp } = messageProperties
+
+    const comments = await DiscourseService.fetchAllComments(Number(GATSBY_DISCOURSE_CONNECT_THREAD))
+    const timeWindow = new Date(new Date().getTime() - MESSAGE_TIMEOUT_TIME)
+
+    const filteredComments = comments.filter((comment) => new Date(comment.created_at) > timeWindow)
+    const validComment = filteredComments.find((comment) => {
+      const addressRegex = new RegExp(address, 'i')
+      const dateRegex = new RegExp(timestamp, 'i')
+
+      return addressRegex.test(comment.cooked) && dateRegex.test(comment.cooked)
+    })
+
+    if (validComment) {
+      const signatureRegex = /0x([a-fA-F\d]{130})/
+      const signature = '0x' + validComment.cooked.match(signatureRegex)?.[1]
+      const recoveredAddress = recoverAddress(hashMessage(formatValidationMessage(address, timestamp)), signature)
+
+      if (recoveredAddress.toLowerCase() !== user) {
+        throw new Error('Invalid signature')
+      }
+
+      await UserModel.createConnection(user, validComment.user_id)
+      clearValidationInProgress(user)
+    }
+
+    return {
+      valid: !!validComment,
+    }
+  } catch (error) {
+    throw new Error("Couldn't validate the user. " + error)
+  }
+}
+
+async function isValidated(req: Request) {
+  const address = req.params.address
+  if (!isEthereumAddress(address)) {
+    return false
```

We should throw an error here